server/neptune/workflows/internal/lyft/audit_workflow_runner_wrapper.go

```diff
@@ -0,0 +1,61 @@
+package lyft
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/revision/queue"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/job"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/root"
+	"go.temporal.io/sdk/workflow"
+)
+
+type auditActivities interface {
+	AuditJob(ctx context.Context, request activities.AuditJobRequest) error
+}
+
+type AuditWorkflowRunnerWrapper struct {
+	Activity auditActivities
+	queue.TerraformWorkflowRunner
+}
+
+func (w *AuditWorkflowRunnerWrapper) Run(ctx workflow.Context, deploymentInfo terraform.DeploymentInfo) error {
+	if err := w.emit(ctx, job.Running, deploymentInfo); err != nil {
+		return errors.Wrap(err, "emitting atlantis job event")
+	}
+
+	result := w.TerraformWorkflowRunner.Run(ctx, deploymentInfo)
+
+	if result != nil {
+		if err := w.emit(ctx, job.Failure, deploymentInfo); err != nil {
+			logger.Error(ctx, errors.Wrap(err, "failed to emit atlantis job event").Error())
+		}
+	} else {
+		if err := w.emit(ctx, job.Success, deploymentInfo); err != nil {
+			logger.Error(ctx, errors.Wrap(err, "failed to emit atlantis job event").Error())
+		}
+	}
+
+	return result
+}
+
+func (w *AuditWorkflowRunnerWrapper) emit(ctx workflow.Context, state job.State, deploymentInfo terraform.DeploymentInfo) error {
```

Comment:
should this be the `root.DeploymentInfo` version to avoid converting here?

Reply:
Do you mean passing `root.DeploymentInfo` to the wrapper? This wrapper is called after we pop `terraform.DeploymentInfo` from the queue so we have to convert it eventually, unless I'm missing something!

Reply:
We use `terraform.DeploymentInfo` inside the deploy workflow and create a `root.DeploymentInfo` when we have to pass it to an activity. The reason we do that is because it creates a cyclic dependency since the terraform package in the deploy workflow calls the activities. So we cannot use `terraform.DeploymentInfo` in the activities package unless we can isolate the activity calls in the terraform package from the `terraform.DeploymentInfo` object.  

I initially was in favor of moving the `terraform.DeploymentInfo` object outside of the terraform package so that it can be shared between the workflow code and activities code but @nish suggested we keep the workflow and activities models isolated. So, we decided to create `root.DeploymentInfo` which is used when passing the original deployment info object to the activities. 

I have added the missing version field for now.